### PR TITLE
fix check resource exists

### LINKED ISSUE #347: [BUG]When error is returned, it does not necessarily mean that the resource does not exist
https://github.com/oam-dev/kubevela/blob/6ede8601b4025be90b3b3578f913815b26b99276/pkg/commands/util/init.go#L46-L50

https://github.com/oam-dev/kubevela/blob/6ede8601b4025be90b3b3578f913815b26b99276/pkg/commands/util/init.go#L61-L63

When error is returned, it does not necessarily mean that the resource does not exist.

should use `errors.IsNotFound` to check.

### PR DESCRIPTION
**Pull request by silenceper**
fix #347

### CODE REVIEW

File: pkg/commands/util/init.go [hunk 1]
```diff
@@ -43,10 +44,18 @@ func GetKubeConfig() string {
 	return filepath.Join(HomeDir(), ".kube", "config")
 }
 
-func IsNamespaceExist(c client.Client, namespace string) bool {
+// DoesNamespaceExist check namespace exist
```
Comment (zzxwill) [inline pkg/commands/util/init.go:47]: I prefer to keep the previous logic as it is. `IsNamespaceExist` or `DoesNamespaceExist`, as a very simple function to check whether the namespace is existed, we don't care the error message a lot. IMHO, returning a true or false is better.

Comment (silenceper) [reply pkg/commands/util/init.go:47]: How to deal with the error situation, such as a network error may occur, choose to ignore it?

File: pkg/commands/util/init.go [hunk 2]
```diff
@@ -43,10 +44,17 @@ func GetKubeConfig() string {
...
```
[the remark this answers is not in the document]
Comment (silenceper) [reply pkg/commands/util/init.go:47]: no problem.

**Comment by silenceper** [comment]
- add tests
- fix `DoesCRDExist`, It may be more appropriate to check exists based on the crd name(This method is not used)

File: pkg/commands/util/init_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var (
+	scheme *runtime.Scheme
+)
+
+func init() {
```
Comment (wonderflow) [inline pkg/commands/util/init_test.go:21]: we should not use `init` in test package

Comment (silenceper) [reply pkg/commands/util/init_test.go:21]: 😅 I refer to the implementation here and keep it consistent.
I will remove it.

https://github.com/oam-dev/kubevela/blob/64d77656cf5159ecc72493710643de76461c95f7/pkg/controller/dependency/install_test.go#L46-L51

Comment (wonderflow) [reply pkg/commands/util/init_test.go]: yes, we should remove both of them. Thanks for pointing out.

File: pkg/controller/dependency/install.go [hunk 4]
```diff
@@ -84,9 +84,15 @@ func installHelmChart(client client.Client, chart []byte, log logr.Logger) error
 	log.Info("install helm char", "chart name", helmChart.Name)
 	// create the namespace
 	if helmChart.Namespace != types.DefaultAppNamespace {
-		if len(helmChart.Namespace) > 0 && !cmdutil.IsNamespaceExist(client, helmChart.Namespace) {
-			if err = cmdutil.NewNamespace(client, helmChart.Namespace); err != nil {
-				return errors.Wrap(err, "failed to create the namespace")
+		if len(helmChart.Namespace) > 0 {
+			exist, err := cmdutil.DoesNamespaceExist(client, helmChart.Namespace)
+			if err != nil {
+				return nil
```
Comment (wonderflow) [inline pkg/controller/dependency/install.go:90]:
```suggestion
				return err
```
？

Comment (silenceper) [reply pkg/controller/dependency/install.go:90]: sorry, typo

**Review by wonderflow — APPROVED** [review]

**Review by ryanzhang-oss — APPROVED** [review]
LGTM

File: pkg/commands/util/init.go (hunk 1 above)
Comment (zzxwill) [reply pkg/commands/util/init.go:47]: +1 for your change. Thanks.

**Pull request merged**